p2panda-rs/src/next/operation/verified_operation.rs

```diff
@@ -0,0 +1,144 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+use std::hash::Hash as StdHash;
+
+use crate::next::entry::Entry;
+use crate::next::identity::Author;
+use crate::next::operation::traits::AsVerifiedOperation;
+use crate::next::operation::{Operation, OperationId};
+
+/// An operation which has been encoded and published on a signed entry.
+///
+/// Contains the values of an operation as well as its author and id. This
+/// [operation id][OperationId] is only available on [`VerifiedOperation`] and not on
+/// [`Operation`] because it is derived from the hash of the signed entry an operation is encoded
+/// on.
+#[derive(Debug, Clone)]
+pub struct VerifiedOperation {
+    /// Identifier of the operation.
+    pub(crate) operation_id: OperationId,
+
+    /// Operation, which is the payload of the entry.
+    pub(crate) operation: Operation,
+
+    /// Entry which was used to publish this operation.
+    pub(crate) entry: Entry,
```

Comment:
I understand that a `VerifiedOperation` is now _actually_ verified ;-p but could you expand on the reason for having the whole entry accessible here now as well.

Reply:
Good question! No clear reason, other than we have the object, .. I think we can totally change this if it doesn't make sense